// frontend/components/data.ts
// data.ts
export const events = [
  {
    id: 1,
    title: "Black Friday and Cyber Monday:",
    description:
      "Offer significant discounts and flash sales on a variety of products.",
    image1: "/maincard/watch.png",
    image2: "/maincard/headset.png",
    background: "bg-mo",
    textStyle:
      "absolute flex flex-col  h-[60%] top-[20%] justify-between left-[84px] w-[340px] text-bg",
    image1Style: "absolute top-[-50px] left-[35%] flex   w-[280px]",
    image2Style: "absolute top-[50px] right-[84px] flex   w-[420px]",
    buttonStyle: "bg-bl text-bg",
  },
  {
    id: 2,
    title: "Holiday Goods",
    description: "Celebrate the season with amazing offers!",
    image1: "/maincard/vr.png",
    image2: "/maincard/phone.png",
    background: "bg-mb",
    textStyle:
      "absolute left-[84px]  h-[60%] top-[20%] flex flex-col justify-between w-[340px] text-bg",
    image1Style: "absolute top-[-50px] left-[34%] flex  w-[470px]",
    image2Style: "absolute top-[40px] right-[84px] flex  w-[230px]",
    buttonStyle: "bg-bg text-bl",
  },
  {
    id: 3,
    title: "Summer Sale",
    description: "Enjoy summer vibes with great discounts!",
    image1: "/maincard/shirts.png",
    image2: "/maincard/jacket.png",
    background: "bg-mg",
    textStyle:
      "absolute left-[84px]  h-[60%] top-[20%] flex flex-col justify-between w-[340px] text-bg",
    image1Style: "absolute top-[10px] left-[25%] flex w-[520px]",
    image2Style: "absolute top-[20px] right-[6%] flex w-[300px]",
    buttonStyle: "bg-mo text-bg",
  },
];
